### ISSUE TITLE
Library augmentations

### ISSUE THREAD

**Issue by jakemac53** [dart-lang/language#4154]
Tracking issue, see feature spec here https://github.com/dart-lang/language/blob/main/working/augmentation-libraries/feature-specification.md

**Comment by Lootwig** [comment]
While this is under development, is there any way to tell which version of the spec corresponds to a particular version of the dart SDK? (in particular, a quick way to figure out that sdk 3.7.0-106.0.dev was created when `augment super()` and `import augment 'other.dart'` existed)

**Comment by jakemac53** [comment]
This feature is under active development, the implementations won't perfectly match the spec at any given state and also won't even match each other (there are two implementations at least, the CFE and analyzer).

In particular both of those teams are undergoing some large refactors to support enhanced parts at the moment, and _then_ they will start actively implementing augmentations further. But, it will be quite some time before they are generally usable and stable.

**Comment by albinpk** [comment]
I’m trying to experiment with Dart augmentations, but I haven’t been able to get them working.

I’m currently on the Dart dev channel.
Could someone clarify the correct steps to enable or use augmentations?

**Comment by ykmnkmi** [comment]
Last time I checked, it was under the `macro` experiment. It was `3.9.x`.

**Comment by albinpk** [comment]
Yeah, I did try `macros` when they were in development. Now that macros are cancelled, I’m experimenting with `augmentations` only. However, I’m unable to get them running on the dev channel. I also had to enable the `macros` experimental flag; otherwise, I get lint errors.

**Dart version:**
```bash
$ dartdev --version
Dart SDK version: 3.12.0-154.0.dev (dev) (Mon Feb 16 12:09:02 2026 -0800) on "macos_arm64" 
```

**analysis_options.yaml:**

```yaml
include: package:lints/recommended.yaml

analyzer:
  enable-experiment:
    - augmentations
    - macros
```

**main.dart:**
```dart
void main() {
  print(A().x);
}

class A {}

augment class A {
  int x = 1;
}
```

**Command used:**

```bash
$ dartdev run --enable-experiment=augmentations,macros lib/main.dart

lib/main.dart:7:1: Error: Variables must be declared using the keywords 'const', 'final', 'var' or a type name.
Try adding the name of the type of the variable or the keyword 'var'.
augment class A {
^^^^^^^
lib/main.dart:7:1: Error: Expected ';' after this.
augment class A {
^^^^^^^
lib/main.dart:7:15: Error: 'A' is already declared in this scope.
augment class A {
              ^
lib/main.dart:5:7: Context: Previous declaration of 'A'.
class A {}
      ^
```

Did I miss anything? Please let me know if there’s any additional setup or flag required.

**Comment by lrhn** [comment]
Your language version must be 3.12 (newest version supported by the SDK) for the experiment flag to have any effect. Check that your `pubspec.yaml` has an SDK version of `^3.12.0-0` or higher.

(Even then, there is no promise that experiments actually work.)

**Comment by albinpk** [comment]
I created the project using Dart 3.12

Here is my `pubspec.yaml`:

```yaml
name: augmentation
version: 1.0.0

environment:
  sdk: ^3.12.0-154.0.dev

dev_dependencies:
  lints: ^6.0.0
```
